src/main/java/eu/dissco/backend/service/ElvisService.java

```diff
@@ -0,0 +1,118 @@
+    var specimenList = results.getRight();
+    var inventoryNumbers = specimenList.stream().map(
+        specimen -> new InventoryNumberSuggestion(specimen.getDwcCollectionID(),
+            specimen.getOdsPhysicalSpecimenID(), specimen.getDctermsIdentifier())).toList();
+    return mapper.createObjectNode()
+        .put("total", results.getLeft())
+        .set("inventoryNumbers", mapper.valueToTree(inventoryNumbers));
+  }
+
+  private Pair<Long, List<DigitalSpecimen>> searchElastic(String inventoryNumber, int pageNumber,
+      int pageSize)
+      throws IOException {
+    var map = Map.of("ods:physicalSpecimenID.keyword", List.of(inventoryNumber));
+    return elasticSearchRepository.search(map, pageNumber, pageSize);
+  }
+
+
+  private static ElvisSpecimen buildElvisSpecimen(DigitalSpecimen digitalSpecimen, String url) {
+    var inventoryNumber = digitalSpecimen.getOdsPhysicalSpecimenID();
+    var title = digitalSpecimen.getOdsSpecimenName();
+    var identifier = digitalSpecimen.getDctermsIdentifier();
+    var collectionCode = digitalSpecimen.getDwcCollectionCode();
+    var catalogNumber = digitalSpecimen.getDwcCollectionID();
+    var institutionId = digitalSpecimen.getOdsOrganisationID();
+    var institutionCode = digitalSpecimen.getOdsOrganisationCode();
+    var basisOfRecord = digitalSpecimen.getDwcBasisOfRecord();
+    var uri = url + "/ds/" + digitalSpecimen.getId();
+    var taxonId = getBestTaxonIdentification(digitalSpecimen);
+    String scientificName = null;
+    String scientificNameAuthorship = null;
+    String specificEpithet = null;
+    String family = null;
+    String genus = null;
+    String vernacularName = null;
+    if (taxonId.isPresent()) {
+      scientificName = taxonId.get().getDwcScientificName();
+      scientificNameAuthorship = taxonId.get().getDwcScientificNameAuthorship();
+      specificEpithet = taxonId.get().getDwcSpecificEpithet();
+      family = taxonId.get().getDwcFamily();
+      genus = taxonId.get().getDwcVernacularName();
+      vernacularName = taxonId.get().getDwcVernacularName();
+    }
+    return new ElvisSpecimen(inventoryNumber, title, identifier, collectionCode, catalogNumber, institutionId,
+        institutionCode, basisOfRecord, uri, scientificName, scientificNameAuthorship,
+        specificEpithet, family, genus, vernacularName);
+  }
+
+  private static Optional<TaxonIdentification> getBestTaxonIdentification(
+      DigitalSpecimen digitalSpecimen) {
+    if (digitalSpecimen.getOdsHasIdentifications().isEmpty()
+        || digitalSpecimen.getOdsHasIdentifications().get(0).getOdsHasTaxonIdentifications()
+        .isEmpty()) {
+      return Optional.empty();
+    }
+    // First, try to find accepted id
+    for (var identification : digitalSpecimen.getOdsHasIdentifications()) {
+      if (Boolean.TRUE.equals(identification.getOdsIsVerifiedIdentification())) {
+        if (identification.getOdsHasTaxonIdentifications().isEmpty()) {
```

Comment:
Do i need to cover the following case: Accepted identification does not contain any taxonIdentifications, but a non-accepted Identification has a taxonIdentification. Seems like that shouldn't happen